vispy/visuals/volume.py

```diff
@@ -367,16 +367,22 @@
     before_loop="""
         float maxval = u_mip_cutoff; // The maximum encountered value
         int maxi = -1;  // Where the maximum value was encountered
+        bool refine = true; // Whether to refine the frag with smaller steps
```

Comment:
Actually, why is this needed? It could still be that the actual best value is *nearby*, so I don't see why we would skip refiniment in these cases. Sure, it improves performance, but then you have a slightly different spatial resolution along the ray cast depending on whether the value was marked as "higher than clim" or not, which is a bit inconsistent.

Reply:
I think you're right. My thinking was if the pixel is going to be pure white (or whatever max val) we can stop all additional work, but this will potentially get the wrong depth slightly. My understanding is a little thin here but I suppose that could be a problem with multiple visuals? In any case this was a pretty small improvement if any so I am happy to remove it.

Reply:
Yeah it's probably minimal, so I would err on the side of not having a different behaviour from normal :+1: